lib/orbit/transformable.js

```diff
@@ -1,121 +1,118 @@
+  // console.log('processTransform', this.id, transform);
 
-  } else {
-    var queue = this._transformationQueue.content;
+  var _this = this;
+  var ops = transform.operations;
+  var result;
+
+  if (this.prepareTransformOperations) {
+    ops = this.prepareTransformOperations(ops);
+  }
 
-    // console.log('transformationFor', operation, queue.length);
+  if (ops.length > 0) {
+    var response = this._transform(ops);
 
-    for (i = 0; i < queue.length; i++) {
-      transformation = queue[i].data;
-      if (transformation.verifyOperation(operation)) {
-        return transformation;
+    if (response) {
+      if (response.then) {
+        return response.then(function(result) {
+          return settleTransform.call(_this, transform, result);
+        });
+      } else if (response instanceof TransformResult) {
+        result = response;
       }
     }
   }
+
+  if (!result) result = new TransformResult();
+
+  return settleTransform.call(this, transform, result);
+}
+
+function queueTransform(transform) {
+  var _this = this;
+
+  var action = this._transformQueue.push({
+    data: transform,
+    process: function() {
+      return processTransform.call(_this, transform);
+    }
+  });
+
+  return action.complete;
 }
 
-function queueTransformation(transformation) {
+function queueTransformResult(transform, result) {
   var _this = this;
 
-  var processor = this._transformationQueue.push({
-    data: transformation,
+  var action = this._transformQueue.push({
+    data: transform,
     process: function() {
-      return transformation.process();
+      return settleTransform.call(_this, transform, result);
     }
   });
 
-  return processor;
+  return action.complete;
 }
 
 var Transformable = {
   extend: function(object, actions) {
     if (object._transformable === undefined) {
       object._transformable = true;
-      object._transformationQueue = new ActionQueue();
+      object._transformQueue = new ActionQueue();
+      object._transformLog = {};
 
       Evented.extend(object);
 
-      object.didTransform = function(operation, inverse) {
-        var normalized = normalize(operation);
-        var transformation = transformationFor.call(this, normalized);
-        if (transformation) {
-          // console.log('Transformable#didTransform - matching transformation found', this.id, normalized, inverse);
-          transformation.pushCompletedOperation(normalized, inverse);
-
-        } else {
-          // console.log('Transformable#didTransform - createTransformation', this.id, normalized, inverse);
-          transformation = new Transformation(this);
-          transformation.pushCompletedOperation(normalized, inverse);
-          queueTransformation.call(this, transformation);
-        }
+      object.didTransform = function() {
+        Orbit.deprecate('`didTransform` has been deprecated. Please call `transformed` instead and supply a TransformResult as an argument.', true);
```

Comment:
Why are we digging for `Orbit.deprecate` here but using `assert` on it's own later?